pkg/cfn/builder/vpc_ipv6_test.go

```diff
@@ -342,6 +343,206 @@ var _ = Describe("IPv6 VPC builder", func() {
 		}))
 	})
 
+	When("private cluster is enabled", func() {
+		It("creates only private IPv6 resources", func() {
+			cfg := cfg.DeepCopy()
+			cfg.PrivateCluster = &api.PrivateCluster{
+				Enabled: true,
+			}
+			cfg.AvailabilityZones = []string{azA, azB}
+			vpcRs = builder.NewIPv6VPCResourceSet(builder.NewRS(), cfg, nil)
+
+			_, subnetDetails, err := vpcRs.CreateTemplate()
+			Expect(err).NotTo(HaveOccurred())
+
+			By("returning the references of public subnets")
+			pubRefs := subnetDetails.PublicSubnetRefs()
+			Expect(pubRefs).To(BeEmpty())
+
+			By("returning the references of private subnets")
+			privRef := subnetDetails.PrivateSubnetRefs()
+			Expect(privRef).To(HaveLen(2))
+			Expect(privRef).To(ContainElement(makePrimitive(builder.PrivateSubnetKey + azBFormatted)))
+			Expect(privRef).To(ContainElement(makePrimitive(builder.PrivateSubnetKey + azBFormatted)))
+
+			vpcTemplate = &fakes.FakeTemplate{}
+			templateBody, err := vpcRs.RenderJSON()
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(json.Unmarshal(templateBody, vpcTemplate)).To(Succeed())
```

Comment:
this PR contains some nice refactoring to make the tests slightly cleaner encase u wanna merge it first https://github.com/weaveworks/eksctl/pull/4378

Reply:
If it means merge conflicts all over the place because of structural changes, no thanks. :D

Reply:
Not to mention that that pr has failing tests. :D

Reply:
not anymore! Also the test was because I merged someones suggesting for changing the wording of the error message! Don't throw shade at me! :joy: :joy:

Reply:
Will merge in :P

Reply:
Uh, this fails so bad now

Reply:
yay this works now. :)